InvenTree/stock/models.py

```diff
@@ -1651,6 +1652,17 @@ def before_delete_stock_item(sender, instance, using, **kwargs):
         child.save()
 
 
+@receiver(post_save, sender=StockItem, dispatch_uid='stock_item_post_save_log')
+def after_save_stock_item(sender, instance: StockItem, **kwargs):
+    """
+    Check if the stock quantity has fallen below the minimum threshold of part. If yes, notify the users who have
+    starred the part
+    """
+
+    if instance.quantity <= instance.part.minimum_stock:
```

Comment:
Refer to my note above regarding "StockItem quantity" vs "Part quantity".

Here you should be checking something along the lines of:

```python
if instance.part.total_stock <= instance.part.minimum_stock:
```

Although I would suggest that this is written as a method of the `Part` model and called here.

Reply:
I would suggest that you add a method to the `Part` model called `check_low_stock()` which you can then call from the post_save hook